covid19-dashboard_NEW/main.py

```diff
@@ -0,0 +1,360 @@
+                                              populationType="Person",
+                                              observationDate="2018"),
+                                         api_key=API_KEY,
+                                         compress=True)
+
+    # Merge US State and US County into one response.
+    response: list = state_response['places'] + county_response['places']
+
+    output: dict = {}
+
+    for place in response:
+        # Some geoIds return None if there is no data.
+        # Make sure there is data first.
+        if 'place' not in place and not place['place']:
+            continue
+        geo_id: str = place['place']
+
+        # The census data can be found in place["observations"].
+        if "observations" not in place and not place["observations"]:
+            continue
+
+        # Search for CensusACS5yrSurvey in observations.
+        for observation in place["observations"]:
+            # If you find the census data
+            if observation["measurementMethod"] == "CensusACS5yrSurvey":
+                # Make sure measuredValue has a real value.
+                # It is the population.
+                if 'measuredValue' not in observation:
+                    continue
+                # Store the population.
+                output[geo_id] = observation['measuredValue']
+
+    # NYT exceptions.
+    # New York City.
+    output['geoId/3651000'] = 8_399_000
+    # Kansas City.
+    output['geoId/2938000'] = 491_918
+
+    return output
+
+
+def get_places(return_counties=False):
+    """
+    Returns a list of places in the United States.
+    :param return_counties: whether to return information
+    for counties+states or just states.
+    :return: geoId -> (name, containedIn, typeOfRegion)
+    """
+
+    # Get US State data.
+    response: dict = send_request(DC_SERVER + "node/places-in", {
+        "dcids": ["country/USA"],
+        "placeType": "State"
+    },
+                                  api_key=API_KEY)
+
+    # Get the names for the US State data.
+    state_dcids: list = [x['place'] for x in response]
+    response: dict = send_request(DC_SERVER + "node/property-values", {
+        "dcids": state_dcids,
+        "property": "name"
+    },
+                                  api_key=API_KEY)
+    states: dict = {}
+
+    # For every geoId, store (name, containedIn, typeOfRegion).
+    for geo_id in response:
+        if 'out' in response[geo_id] and response[geo_id]['out']:
+            if 'value' in response[geo_id]['out'][0]:
+                # Store the data for the geoId.
+                name = response[geo_id]['out'][0]['value']
+                states[geo_id] = (name, 'country/USA', "State")
+
+    # Get US County data.
+    response: dict = send_request(DC_SERVER + "node/places-in", {
+        "dcids": state_dcids,
+        "placeType": "County"
+    },
+                                  api_key=API_KEY)
+
+    # Keep track of what county each state belongs to.
```

Comment:
counties belong to states, not the other way round :)

Reply:
Whoops thanks!